jsapp/js/dataInterface.d.ts

```diff
@@ -7,6 +7,45 @@ interface FailResponse {
   statusText: string
 }
 
+interface SubmissionAttachment {
+  download_url: string
+  download_large_url: string
+  download_medium_url: string
+  download_small_url: string
+  mimetype: string
+  filename: string
+  instance: number
+  xform: number
+  id: number
+}
+
+interface SubmissionResponse {
+  [questionName: string]: any
+  __version__: string
+  _attachments: SubmissionAttachment[]
```

Comment:
Found a bug, at this line:
https://github.com/kobotoolbox/kpi/blob/80fbec7eab5bc9d16f97e662a56c43be6a64bf67/jsapp/js/components/submissions/table.es6#L944

it has to do with the `row.original` having `_attachments` rather than `row` itself. I think we should change `getMediaAttachment` param to an object having `{_attachment: any}` instead of `SubmissionResponse`, as in this case it would not be that type

I commited a fix here: https://github.com/kobotoolbox/kpi/pull/3458/commits/3e72eaa8d5023b68235c735135d806582066cbc4

Reply:
Thanks for the fix!

Reply:
> I think we should change `getMediaAttachment` param to an object having `{_attachment: any}` instead of `SubmissionResponse,` as in this case it would not be that type
I don't think I understand - in all two of the `getMediaAttachment` calls we are passing the `row.original`, which is `SubmissionResponse` in both cases

Reply:
> > I think we should change `getMediaAttachment` param to an object having `{_attachment: any}` instead of `SubmissionResponse,` as in this case it would not be that type
> 
> I don't think I understand - in all two of the `getMediaAttachment` calls we are passing the `row.original`, which is `SubmissionResponse` in both cases

You're right. I didn't notice `row.original` was the submission response. I confused it with just `row` :P